### DISCUSSION TITLE
How to make a curve collidable

### DISCUSSION THREAD

**Discussion by flaming-flamingo** [AlmasB/FXGL#1366] [General]
Hello. I am currently working on a game which requires an entity to be launched off a ramp, starting from a high point and descending until it reaches an upward slope. However, I have had to resort to the use of triangles and rectangles when using a QuadCurve would be more helpful, but my attemps at making the curve collidable are innefective. how can I make it so that it is collidable?

**Comment by AlmasB** [comment]
> my attemps at making the curve collidable

Hi, what is the approach you use now to make it collidable? There is no built-in curve collision detection, but you should be able to approximate the curve with a polygon and outsource the checks to the embedded box2d physics engine. Here's a [sample](https://github.com/AlmasB/FXGL/blob/10ee77c01791be7c9e6c81ed1ccf187b651e07b4/fxgl-samples/src/main/java/intermediate/PhysicsPlaygroundSample.java) that shows some of that.
